beacon-chain/core/light-client/lightclient.go

```diff
@@ -611,64 +454,48 @@ func getExecutionPayloadHeaderDeneb(block interfaces.ReadOnlySignedBeaconBlock)
 	} else if err != nil {
 		return nil, errors.Wrap(err, "could not get withdrawals root")
 	}
-
-	execution := &enginev1.ExecutionPayloadHeaderDeneb{
-		ParentHash:       payloadInterface.ParentHash(),
-		FeeRecipient:     payloadInterface.FeeRecipient(),
-		StateRoot:        payloadInterface.StateRoot(),
-		ReceiptsRoot:     payloadInterface.ReceiptsRoot(),
-		LogsBloom:        payloadInterface.LogsBloom(),
-		PrevRandao:       payloadInterface.PrevRandao(),
-		BlockNumber:      payloadInterface.BlockNumber(),
-		GasLimit:         payloadInterface.GasLimit(),
-		GasUsed:          payloadInterface.GasUsed(),
-		Timestamp:        payloadInterface.Timestamp(),
-		ExtraData:        payloadInterface.ExtraData(),
-		BaseFeePerGas:    payloadInterface.BaseFeePerGas(),
-		BlockHash:        payloadInterface.BlockHash(),
-		TransactionsRoot: transactionsRoot,
-		WithdrawalsRoot:  withdrawalsRoot,
-	}
-
-	return execution, nil
+	return withdrawalsRoot, nil
 }
 
-func ComputeTransactionsRoot(payload interfaces.ExecutionData) ([]byte, error) {
-	transactionsRoot, err := payload.TransactionsRoot()
-	if errors.Is(err, consensus_types.ErrUnsupportedField) {
-		transactions, err := payload.Transactions()
+func BlockToLightClientHeader(block interfaces.ReadOnlySignedBeaconBlock) (*ethpbv2.LightClientHeaderContainer, error) {
+	switch block.Version() {
+	case version.Altair, version.Bellatrix:
+		altairHeader, err := BlockToLightClientHeaderAltair(block)
 		if err != nil {
-			return nil, errors.Wrap(err, "could not get transactions")
+			return nil, errors.Wrap(err, "could not get altair header")
```

Comment:
```suggestion
			return nil, errors.Wrap(err, "could not get header")
```
Returning an error that says `altair` for a Bellatrix block will be confusing